src/main/java/jenkins/bouncycastle/api/PEMEncodable.java

```diff
@@ -487,5 +491,25 @@ private static String encodeBase64(@Nonnull byte[] data) {
         return Base64.decode(data);
     }
 
+    /**
+     * Registers bouncy castle on the slave JVM
+     * 
+     * @param channel to the slave
+     * @return result of the operation
+     * @throws Exception if there is a problem registering bouncycastle
+     */
+    public static boolean registerBCOnSlave(@Nonnull Channel channel) throws Exception {
+        if (channel.getProperty(BOUNCYCASTLE_REGISTERED) == null) {
+            // pre-loading the bouncyclastle jar to make sure the JVM reconizes the signature
+            channel.preloadJar(PEMEncodable.class.getClassLoader(), BouncyCastleProvider.class);
```

Comment:
I think so, AFIK you need a class loader that know where the correct class is loaded from, in this case BouncyCastleProvider.class